src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -8514,5 +8508,293 @@ public void M()
+  .maxstack  1
+  .locals init (int V_0, //x
+                int V_1) //y
+  IL_0000:  ldarg.0
+  IL_0001:  brtrue.s   IL_0016
+  IL_0003:  ldarg.1
+  IL_0004:  brtrue.s   IL_000e
+  IL_0006:  ldc.i4.s   50
+  IL_0008:  stloc.0
+  IL_0009:  ldc.i4.s   60
+  IL_000b:  stloc.1
+  IL_000c:  br.s       IL_001c
+  IL_000e:  ldc.i4.s   30
+  IL_0010:  stloc.0
+  IL_0011:  ldc.i4.s   40
+  IL_0013:  stloc.1
+  IL_0014:  br.s       IL_001c
+  IL_0016:  ldc.i4.s   10
+  IL_0018:  stloc.0
+  IL_0019:  ldc.i4.s   20
+  IL_001b:  stloc.1
+  IL_001c:  ldloc.0
+  IL_001d:  call       ""void System.Console.Write(int)""
+  IL_0022:  ldloc.1
+  IL_0023:  call       ""void System.Console.Write(int)""
+  IL_0028:  ret
+}");
+        }
+
+        [Fact]
+        public void AssigningConditional_MixedDeconstruction()
+        {
+            string source = @"
+using System;
+
+class C
+{
+    static void Main()
+    {
+        M(true, false);
+        M(false, true);
+        M(false, false);
+    }
+
+    static void M(bool b1, bool b2)
+    {
+        (var x, long y) = b1 ? (10, 20) : b2 ? (30, 40) : (50, 60);
+        Console.Write(x);
+        Console.Write(y);
+    }
+}
+";
+
+            var comp = CompileAndVerify(source, options: TestOptions.ReleaseExe, expectedOutput: "102030405060");
+            comp.VerifyDiagnostics();
+            comp.VerifyIL("C.M", @"
+{
+  // Code size       50 (0x32)
+  .maxstack  1
+  .locals init (int V_0, //x
+                long V_1, //y
+                long V_2)
+  IL_0000:  ldarg.0
+  IL_0001:  brtrue.s   IL_001c
+  IL_0003:  ldarg.1
+  IL_0004:  brtrue.s   IL_0011
+  IL_0006:  ldc.i4.s   60
+  IL_0008:  conv.i8
+  IL_0009:  stloc.2
+  IL_000a:  ldc.i4.s   50
+  IL_000c:  stloc.0
+  IL_000d:  ldloc.2
+  IL_000e:  stloc.1
+  IL_000f:  br.s       IL_0025
+  IL_0011:  ldc.i4.s   40
+  IL_0013:  conv.i8
+  IL_0014:  stloc.2
+  IL_0015:  ldc.i4.s   30
+  IL_0017:  stloc.0
+  IL_0018:  ldloc.2
+  IL_0019:  stloc.1
+  IL_001a:  br.s       IL_0025
+  IL_001c:  ldc.i4.s   20
+  IL_001e:  conv.i8
+  IL_001f:  stloc.2
+  IL_0020:  ldc.i4.s   10
+  IL_0022:  stloc.0
+  IL_0023:  ldloc.2
+  IL_0024:  stloc.1
+  IL_0025:  ldloc.0
+  IL_0026:  call       ""void System.Console.Write(int)""
+  IL_002b:  ldloc.1
+  IL_002c:  call       ""void System.Console.Write(long)""
+  IL_0031:  ret
+}");
+        }
+
+        [Fact]
+        public void AssigningConditional_SideEffects()
+        {
+            string source = @"
+using System;
+
+class C
+{
+    static void Main()
+    {
+        M(true, false);
+        M(false, true);
+        M(false, false);
+    }
+
+    static int field;
+
+    static ref int SideEffect(int i)
+    {
+        Console.Write(i);
+        return ref field;
+    }
+
+    static void M(bool b1, bool b2)
+    {
+        (SideEffect(1), SideEffect(2)) = b1 ? (10, 20) : b2 ? (30, 40) : (50, 60);
+    }
+}
+";
+
+            var comp = CompileAndVerify(source, options: TestOptions.ReleaseExe, expectedOutput: "121212");
```

Comment:
>121212 [](start = 98, length = 6)

It would be good for `M` to print out the value of `field` after the deconstruction.